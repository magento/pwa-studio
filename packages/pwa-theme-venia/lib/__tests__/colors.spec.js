const { declareColors, getColors } = require('../colors');

describe('declareColors()', () => {
    test('returns the correct values with default parameters', () => {
        const colorDeclarations = declareColors();

        expect(colorDeclarations).toMatchSnapshot();
    });

    test('handles non-nested definition', () => {
        const data = {
            white: '255 255 255'
        };

        const colorDeclarations = declareColors(data);

        expect(colorDeclarations).toMatchInlineSnapshot(`
            Object {
              "--color-white": "255 255 255",
            }
        `);
    });
});

describe('getColors()', () => {
    test('returns the correct values with default parameters', () => {
        const colorConfig = getColors();

        expect(colorConfig).toMatchSnapshot();
    });

    describe('returns property functions which return the correct format with', () => {
        const data = {
            brand: {
                400: '38 128 235',
                500: '20 115 230',
                600: '13 102 208',
                700: '9 90 186'
            },
            black: '0 0 0'
        };

        const colorConfig = getColors(data);
        const { brand, black } = colorConfig;

        test('no opacity args', () => {
            const opacityArgs = {};
            expect(brand[400](opacityArgs)).toMatchInlineSnapshot(
                `"rgb(var(--color-brand-400))"`
            );
        });

        test('opacity value given', () => {
            const opacityArgs = {
                opacityValue: 0.5
            };
            expect(brand[500](opacityArgs)).toMatchInlineSnapshot(
                `"rgb(var(--color-brand-500) / 0.5)"`
            );
        });

        test('opacity variable given', () => {
            const opacityArgs = {
                opacityVariable: '--my-opacity'
            };
            expect(black(opacityArgs)).toMatchInlineSnapshot(
                `"rgb(var(--color-black) / var(--my-opacity, 1))"`
            );
        });
    });
});
