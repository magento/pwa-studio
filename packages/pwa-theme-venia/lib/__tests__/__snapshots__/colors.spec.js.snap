// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declareColors() returns the correct values with default parameters 1`] = `
Object {
  "--color-black": "0 0 0",
  "--color-gray-100": "244 245 245",
  "--color-gray-200": "234 235 235",
  "--color-gray-300": "223 225 226",
  "--color-gray-400": "181 184 186",
  "--color-gray-50": "255 255 255",
  "--color-gray-500": "149 154 157",
  "--color-gray-600": "118 123 127",
  "--color-gray-700": "84 93 99",
  "--color-gray-75": "250 250 250",
  "--color-gray-800": "51 63 71",
  "--color-gray-900": "23 32 38 ",
  "--color-neutral-d1": "83 89 101",
  "--color-neutral-d2": "24 28 36",
  "--color-neutral-l0": "255 255 255",
  "--color-neutral-l1": "229 229 231",
  "--color-neutral-l2": "181 183 188",
  "--color-neutral-m1": "145 149 156",
  "--color-primary-d1": "31 57 255",
  "--color-primary-l1": "61 132 255",
  "--color-primary-l2": "51 109 255",
  "--color-primary-m1": "41 84 255",
  "--color-white": "255 255 255",
}
`;

exports[`getColors() returns the correct values with default parameters 1`] = `
Object {
  "black": [Function],
  "gray": Object {
    "100": [Function],
    "200": [Function],
    "300": [Function],
    "400": [Function],
    "50": [Function],
    "500": [Function],
    "600": [Function],
    "700": [Function],
    "75": [Function],
    "800": [Function],
    "900": [Function],
  },
  "neutral": Object {
    "d1": [Function],
    "d2": [Function],
    "l0": [Function],
    "l1": [Function],
    "l2": [Function],
    "m1": [Function],
  },
  "primary": Object {
    "d1": [Function],
    "l1": [Function],
    "l2": [Function],
    "m1": [Function],
  },
  "white": [Function],
}
`;
